File: Sources/windows/yage/models/species.py
from typing import List
from yage.models.animations import EntityAnimation


class Species:
    def __init__(
        self,
        species_id: str,
        animations: List[EntityAnimation] = [],
        capabilities: List[str] = [],
        drag_path: str = 'drag',
        fps: float = 10,
        movement_path: str = 'walk',
        speed: float = 1,
        tags: List[str] = [],
        z_index: float = 0
    ):
        self.id = species_id
        self.animations = animations
        self.capabilities = capabilities
        self.drag_path = drag_path
        self.fps = fps
        self.movement_path = movement_path
        self.speed = speed
        self.tags = tags
        self.z_index = z_index

    def __eq__(self, other: 'Species'):
        if other is None:
            return False
        return self.id == other.id

    def __hash__(self):
        return hash(self.id)

    def __repr__(self):
        return self.id


SPECIES_AGENT = Species(species_id="agent")
SPECIES_HOTSPOT = Species(species_id="hotspot")

File: Sources/windows/App/Sources/Species/Species.cs
namespace App
{
    public class Species
    {
        public string Id { get; set; }
        public string DragPath { get; set; }
        public string MovementPath { get; set; }
        public int ZIndex { get; set; }
        public double Speed { get; set; }
        public double Scale { get; set; }
        public List<SpeciesAnimation> Animations { get; set; }
        public List<string> Capabilities { get; set; }

        public Species(string id, double speed, double scale)
        {
            Id = id;
            Speed = speed;
            Scale = scale;
            DragPath = "drag";
            MovementPath = "walk";
            ZIndex = 0;
            Animations = new List<SpeciesAnimation>();
            Capabilities = new List<string>();
        }
    }
}
